porch/api/go.mod

```diff
@@ -3,10 +3,11 @@ module github.com/GoogleContainerTools/kpt/porch/api
 go 1.17
 
 require (
-	k8s.io/apimachinery v0.23.5
-	k8s.io/client-go v0.23.5
-	k8s.io/code-generator v0.23.5
-	k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65
+	github.com/GoogleContainerTools/kpt v1.0.0-beta.16
```

Comment:
Is this dependency okay?

Reply:
I agree this might cause issues. I don't necessarily think it is a problem that the porch API has a dependency on kpt since porch is built upon the kpt concepts, and the `UpstreamLock` is a pretty important part of the kpt api.

I think splitting the kpt library from the kpt cli as I suggest in https://github.com/GoogleContainerTools/kpt/issues/3297#issuecomment-1151462512 would address the problem with any circular dependencies. The kpt cli can have dependencies on the porch api, but the kpt library should not.

Reply:
hmm...if this change will be blocked on the pkg restructuring, then we will have to go back to the plan of doing this step in the CLI, right ?

Reply:
An intermediate solution would be to define a duplicate `UpstreamLock` type in the porch API. In the client side I can do a trivial translation, and leave a TODO in the code that the duplicate type should go away with https://github.com/GoogleContainerTools/kpt/issues/3297#issuecomment-1151462512.

Reply:
I like option 1 of duplicating `UpstreamLock` because it will unblock the `pkg update`. It helps us unblock update overall.

We might hit some unknowns in pkg restructuring or it might take a bit longer risking `pkg update` timelines.

Reply:
Cool, I updated the PR to create duplicate `UpstreamLock` type.